exec/java-exec/src/main/resources/drill-module.conf

```diff
@@ -115,7 +115,8 @@ drill.exec: {
       text: {
         buffer.size: 262144,
         batch.size: 4000
-      }
+      },
+      recursive_listing_max_size: 10000
```

Comment:
Could you please make this limit optional?

Reply:
My 2 cents is that limits ideally should be set by default to a sensible level. For Drill 2.0.0, enforcing that some sort of limit is set would be something that I'd support. For Drill 1.x, it would not be a good idea to enforce limits by default but supporting them optionally would be useful (to avoid introducing changes that might force users to tune configs in a minor release).

Reply:
Yes, the default value should be adjusted. For the big data world, thousands of files are quite a small amount. For non-parquet files FileStatus is small, so it shouldn't cause large pressure on memory. For parquet files, it would be good to provide the functionality to disable reading metadata for planning and use it only during execution to avoid issues with huge files amount.

Reply:
A limit of 0 (or less) now means no limit and the default, for this PR, is now 0.